src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py

```diff
@@ -276,3 +276,16 @@ def load_arguments(self, _):
     with self.argument_context('account management-group update') as c:
         c.argument('display_name', options_list=['--display-name', '-d'])
         c.argument('parent_id', options_list=['--parent', '-p'])
+
+    with self.argument_context('rest') as c:
+        c.argument('method', arg_type=get_enum_type(['head', 'get', 'put', 'post', 'delete', 'options', 'patch'], default='get'),
+                   help='HTTP request method')
+        c.argument('url', help="request url")
+        c.argument('headers', nargs='+', help="Space-separated appsettings in KEY=VALUE format or JSON string. Use @{file} to load from a file")
+        c.argument('uri_parameters', nargs='+', help='Space-separated appsettings in KEY=VALUE format or JSON string. Use @{file} to load from a file')
+        c.argument('skip_authorization_header', action='store_true', help='do not auto append "Authorization" header')
+        c.argument('body', help='request body')
+        c.argument('output_file', help='save response payload to a file')
+        c.argument('resource', help='Resource which CLI would acquire token so to put on the "Authorization" header. '
```

Comment:
This can possibly reworded / made more concise.

Also should this be called `--endpoint`, `--auth-resource` or `--resource-type` or something?
An example using this parameter would be wonderful, as this option might be one of the least straightforward, even those it seems to be simply the base url or a prefix of the request url.

```suggestion
        c.argument('resource', options_list=['--endpoint'], help='Endpoint to use when retrieving the auth token for the "AUTHORIZATION" header. Defaults to an endpoint in `az cloud show`. Values from `az cloud show --query "endpoints"`'
```

Reply:
Proposed re-wording
```suggestion
        c.argument('resource', help='Resource for which the CLI should acquire a token to be used for authorization. The token will be place in the "Authorization" header.  Values from `az cloud show --query "endpoints"'  
```
Instead of hardcoding the values from. Maybe we should use populator commands.

Reply:
@yugangw-msft 
Proposed re-wording
```suggestion
        c.argument('resource', help='Resource for which the CLI should acquire a token to be used for authorization. The token will be place in the "Authorization" header.  Values from `az cloud show --query "endpoints"'  
```
Instead of hardcoding the values from. Maybe we should use populator commands.